Modern/ops/src/test/java/org/bonej/ops/ellipsoid/EllipsoidOptimisationStrategyTest.java

```diff
@@ -0,0 +1,307 @@
+				new Vector3d(20.5, 20.5, 20.5), new double[]{40, 40, 40},new NoEllipsoidConstrain());
+		assertNotNull(ellipsoid);
+        assertEquals(9.5,ellipsoid.getRadii()[0],1.0);
+        assertEquals(9.5,ellipsoid.getRadii()[1],1.0);
+        assertEquals(9.5,ellipsoid.getRadii()[2],1.0);
+	}
+
+	private byte[][] getSphere(int radius) {
+		double centrePointCoordinate = 10 + radius + 0.5;
+		Vector3d sphereCentre = new Vector3d(centrePointCoordinate, centrePointCoordinate, centrePointCoordinate);
+
+		int imgDim = 2 * radius + 20;
+		byte[][] sphere = new byte[imgDim][imgDim * imgDim];
+		for (int z = 0; z < imgDim; z++) {
+            for (int y = 0; y < imgDim; y++){
+                for (int x = 0; x < imgDim; x++) {
+                    final Vector3d position = new Vector3d(x + 0.5, y + 0.5, z + 0.5);
+                    position.sub(sphereCentre);
+                    if (position.lengthSquared() <= radius * radius) {
+                        sphere[z][y * imgDim + x] = (byte) 255;
+                    }
+                }
+            }
+		}
+
+		return sphere;
+	}
+
+	/**
+	 * test for {@link EllipsoidOptimisationStrategy#calculateTorque(QuickEllipsoid, Iterable)}
+	 *
+	 * see testFindContactPoints in this file for explanation on what contact points are used
+	 * based on these points, the torque is expected to be zero
+	 *
+	 */
+
+	@Test
+	public void testCalculateTorque() {
+		double[] radii = {1,2.1,3.1};
+		double[] centre = {3,3,2};
+		QuickEllipsoid e = new QuickEllipsoid(radii,centre,new double[][]{{1,0,0},{0,1,0},{0,0,1}});
+
+		double [][] vectors = new double[7][3];
+		vectors[0][0] = 1; //x-direction
+		vectors[1][1] = 1; //y-direction
+		vectors[2][2] = 1; //z-direction
+		vectors[3][0] = -1; //-x-direction
+		vectors[4][1] = -1; //-y-direction
+		vectors[5][2] = -1; //-z-direction
+
+		final byte[][] cubeImage = getCuboidImage();
+
+		//EXECUTE
+		EllipsoidOptimisationStrategy optimisation = new EllipsoidOptimisationStrategy();
+		final ArrayList<double[]> contactPoints = new ArrayList<>();
+		optimisation.findContactPointsForGivenDirections(e, contactPoints, vectors, cubeImage,6,6,6);
+		final double[] torque = EllipsoidOptimisationStrategy.calculateTorque(e, contactPoints);
+
+		assertEquals(0,torque[0],1e-12);
+		assertEquals(0,torque[1],1e-12);
+		assertEquals(0,torque[2],1e-12);
+	}
+
+	/**
+	 * creates a byte array slice representation of a foreground cuboid that touches the outside of an image on one side (z==0).
+	 * @return byte[][] with z index in first index, and xy plane array in second index
+	 */
+	private byte[][] getCuboidImage() {
+		int dimension = 6;
+		final byte[][] cubeImage = new byte[dimension][dimension*dimension];
+
+		for(int x=0;x<dimension;x++) {
+			for (int y = 0; y < dimension; y++) {
+				for (int z = 0; z < dimension; z++) {
+					if (x != 0 && x != 5 && y != 0 && y != 5 && z != 5) //part of z==0 plane is on img boundary and FG
+					{
```

Comment:
I think 
```
if (x == 0 || x == 5 || y == 0 || y == 5 || z == 5)
{
    continue;
}
```
is more clear.

Reply:
May also be faster because you don't have to check all the conditions at runtime: as soon as one condition is satisfied in the 'or' logic you can drop to the statement. Sometimes it gets optimised away by the compiler, other times it makes a difference. It's work checking for performance on inner loops especially ones that are repeated for every pixel, or lots of times per optimisation step.